app/templates/gulpfile.js

```diff
@@ -224,19 +235,118 @@ gulp.task('init-local', function() {
     }
 
     var configString = JSON.stringify(configJSON, null, 2) + '\n';
-    fs.writeFileSync('local.json', configString, { encoding: 'utf8' });
+    fs.writeFileSync('local.json', configString, {
+      encoding: 'utf8'
+    });
   } catch (e) {
     log('failed to write local.json: ' + e.message);
   }
 });
 
 /**
+ * Updates ecosystem.json
+ */
+gulp.task('add-deploy-target', function(done) {
+  log('Update ecosystem.json targets or create new ones!');
+
+  var ecosystem = 'ecosystem.json';
+
+  if (!fs.existsSync(ecosystem)) {
+    try {
+      var configJSON = {};
+      configJSON.deploy = {};
+
+      var configString = JSON.stringify(configJSON, null, 2) + '\n';
+
+      fs.writeFileSync('ecosystem.json', configString, {
+        encoding: 'utf8'
+      });
+    } catch (e) {
+      console.log('failed to write ecosystem.json: ' + e.message);
+    }
+  }
+
+  var properties = fs.readFileSync('deploy/build.properties', {
+    encoding: 'utf8'
+  });
+
+  var name = properties.match(/app-name=(.*)/)[1];
+  var gitUrl = 'https://github.com/';
+  var folderPath = '/space/projects/' + name;
+
+  $.git.exec({
+    args: 'config --get remote.origin.url',
+    quiet: true
+  }, function(err, stdout) {
+    if (!err) {
+      gitUrl = stdout.split('\n')[0];
+    }
+
+    var questions = [{
+      type: 'input',
+      name: 'targetName',
+      message: 'What\'s the name of this deployment target? (If this target exists we\'ll replace it!!!!)',
+      default: 'prod'
+    }, {
+      type: 'input',
+      name: 'key',
+      message: 'Where do you keep your ssh key that\'s used for both the target server and the git repository?',
+      default: '~/.ssh/id_rsa'
+    }, {
+      type: 'input',
+      name: 'username',
+      message: 'Where\'s your username on the target server?',
+      default: username.sync()
+    }, {
+      type: 'input',
+      name: 'hostname',
+      message: 'What\'s the hostname of the target server?',
+      default: 'localhost'
+    }, {
+      type: 'input',
+      name: 'branch',
+      message: 'Which git branch are we deploying?',
+      default: 'master'
+    }, {
+      type: 'input',
+      name: 'gitUrl',
+      message: 'What\'s the url of your git repository?',
+      default: gitUrl
+    }, {
+      type: 'input',
+      name: 'folder',
+      message: 'Where do you want to store the project on your target server?',
+      default: folderPath
+    }];
+
+    gulp.src(ecosystem)
+      .pipe($.prompt.prompt(questions, function(answers) {
+
+        gulp.src(ecosystem)
+          .pipe($.jsonEditor(function(json) {
+            json.deploy[answers.targetName] = {
+              'key': answers.key,
+              'user': answers.username,
+              'host': [answers.hostname],
+              'ref': 'origin/' + answers.branch,
+              'repo': answers.gitUrl,
+              'path': answers.folder,
+              'post-deploy': 'npm install; bower install; gulp build'
```

Comment:
Was wondering if we should use `&&` instead of `;` between the post-deploy commands. That will make post-deploy stop after npm install if that happens to fail.